File: server/src/model/ConstructedObject.java
package model;

import java.util.Date;

public class ConstructedObject extends CoordinateObject {
    private boolean boostBuild;
    private boolean completed;
    private long startBuildTime;
    
    public ConstructedObject(long startBuildTime, boolean boostBuild, boolean completed, int x, int y) {
        super(x, y);
        this.completed = completed;
        this.boostBuild = boostBuild;
        this.startBuildTime = startBuildTime;
    }
    
    public boolean isBoostBuild() {
        return this.boostBuild;
    }
    
    public boolean isCompleted() {
        return this.completed;
    }
    
    public long getStartBuildTime() {
        return this.startBuildTime;    
    }
    
    public void setCompleted() {
        this.completed = true;
    }
    
    public void setBoostBuild() {
        this.boostBuild = true;
    }
    
}

File: client/src/modules/models/ConstructedObject.js
var ConstructedObject = CoordinatedObject.extend({
	completed: false,
	boostBuild: false,
	startBuildTime: 0,
	remainBuildTime: 0,

	ctor: function(startBuildTime, remainBuildTime, boostBuild, completed, coordinate) {
		this._super(coordinate);
		this.remainBuildTime = remainBuildTime;
		this.startBuildTime = startBuildTime;
		this.boostBuild = boostBuild;
		this.completed = completed;
	},

	reduceRemainBuildTime: function (dt) {
		if (this.remainBuildTime > 0) {
			this.remainBuildTime -= dt;
		}
	},

	setBoostBuild: function () {
		this.boostBuild = true;
	}
});
